### PR TITLE
build: enable errorlint and fix issues

### PR DESCRIPTION
**Pull request by lodrem**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Enable linter and fix related issues.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -36,5 +41,5 @@ linters-settings:
   gosec:
     excludes:
       - G114
-  govet:
-    check-shadowing: false
+  goimports:
+    local-prefixes: sigs.k8s.io/cloud-provider-azure
```
Comment (MartinForReal) [inline .golangci.yml:45]: Could you please add a new line here?

File: cmd/acr-credential-provider/main.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ import (
 	"github.com/spf13/cobra"
 	"k8s.io/component-base/logs"
 	"k8s.io/klog/v2"
+
```
Comment (MartinForReal) [inline cmd/acr-credential-provider/main.go:30]: Could you remove the new line and format the code please?

File: pkg/provider/azure_vmssflex_cache.go [hunk 3]
```diff
@@ -197,7 +198,7 @@ func (fs *FlexScaleSet) getNodeVmssFlexID(nodeName string) (string, error) {
 	}
 
 	vmssFlexID, err := getter(nodeName, azcache.CacheReadTypeDefault)
-	if err == cloudprovider.InstanceNotFound {
+	if errors.Is(err, cloudprovider.InstanceNotFound) {
```
Comment (MartinForReal) [inline pkg/provider/azure_vmssflex_cache.go:201]: error might be wrapped with some messages, Do you think we need to invoke errors.As instead?

Comment (lodrem) [reply pkg/provider/azure_vmssflex_cache.go:201]: to log?

File: cmd/acr-credential-provider/main.go (hunk 2 above)
Comment (lodrem) [reply cmd/acr-credential-provider/main.go:30]: the `local-prefixes: sigs.k8s.io/cloud-provider-azure` will group the local import, the new line is expected I think.

**Pull request merged**